src/intersections/segments.jl

```diff
@@ -13,54 +13,57 @@ function intersection(f, s1::Segment{N,T}, s2::Segment{N,T}) where {N,T}
   a, b = s1(0), s1(1)
   c, d = s2(0), s2(1)
 
-  λ₁, λ₂, r, rₐ = intersectparameters(a, b, c, d)
-  # helper function to round values close to x (one or zero)
-  mayberound(λ, x) = isapprox(λ, x, atol=atol(T)) ? x : λ
+  len1, len2 = length(s1), length(s2)
+  b₀ = a + 1/len1 * (b - a) # corresponds to s1(1/length)
+  d₀ = c + 1/len2 * (d - c)
```

Comment:
Why not just write s1(1/length(s1)) ?

Reply:
`segment.jl` (in Polytopes) throws an error since 1 / length(s1) can be larger than 1:

``` julia
function (s::Segment)(t)
  if t < 0 || t > 1
    throw(DomainError(t, "s(t) is not defined for t outside [0, 1]."))
  end
  a, b = s.vertices
  a + t * (b - a)
end
```